solver/llbsolver/ops/source.go

```diff
@@ -73,15 +73,20 @@ func (s *sourceOp) CacheMap(ctx context.Context, g session.Group, index int) (*s
 	}
 
 	dgst := digest.FromBytes([]byte(sourceCacheType + ":" + k))
-
 	if strings.HasPrefix(k, "session:") {
 		dgst = digest.Digest("random:" + strings.TrimPrefix(dgst.String(), dgst.Algorithm().String()+":"))
 	}
 
+	var resolveResponse map[string]string
+	if !strings.HasPrefix(s.op.Source.GetIdentifier(), "local://") {
+		resolveResponse = map[string]string{s.op.Source.GetIdentifier(): k}
+	}
+
 	return &solver.CacheMap{
 		// TODO: add os/arch
-		Digest: dgst,
-		Opts:   cacheOpts,
+		Digest:          dgst,
```

Comment:
I think this digest is not quite what we want. See line 75 where it is already merged with source type. Also I think git adds special things to internal cache key like if the `.git` is kept in source. In here we want the clean object specific hash. Eg. raw image digest, git commit, direct sha of the url.